### PR TITLE
CLOUD-2389 [JDG72] Create templates

### PR DESCRIPTION
**Pull request by vblagoje**
Do *not* integrate yet. Need to confirm that xpass image version is correct, and if not, so I can update the references.

### CODE REVIEW

File: datagrid/datagrid72-image-stream.json [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+                    "version": "1.4.10"
+                } 
+            },
+            "labels": {
+                "xpaas": "1.4.10"
+            },
+            "spec": {
+                "tags": [
+                    {
+                        "name": "1.0",
```
Comment (slaskawi) [inline datagrid/datagrid72-image-stream.json]: I believe we need only `1.0` version tag. Looking at https://issues.jboss.org/browse/CLOUD-2306, you may find (the last comment) that the latest image version is `jboss-datagrid-7/datagrid72-openshift:1.0` (omit the registry name, that will change once RCM push it). So please delete all other tags and leave just 1.0.

File: datagrid/datagrid72-image-stream.json [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+                        "from": {
+                            "kind": "DockerImage",
+                            "name": "registry.access.redhat.com/jboss-datagrid-7/datagrid72-openshift:1.3"
+                        }
+                    }
+                ]
+            }
+        },
+        {
+            "kind": "ImageStream",
```
Comment (slaskawi) [inline datagrid/datagrid72-image-stream.json]: Currently we do not provide the client image. As the matter of fact there are ongoing discussions on PM list whether we need it or not. Please remove it for now.

File: datagrid/datagrid72-postgresql.json [hunk 3]
```diff
@@ -0,0 +1,900 @@
...
+                        "type": "ImageChange",
+                        "imageChangeParams": {
+                            "automatic": true,
+                            "containerNames": [
+                                "${APPLICATION_NAME}"
+                            ],
+                            "from": {
+                                "kind": "ImageStreamTag",
+                                "namespace": "${IMAGE_STREAM_NAMESPACE}",
+                                "name": "jboss-datagrid72-openshift:1.3"
```
Comment (slaskawi) [inline datagrid/datagrid72-postgresql.json]: That should probably be `1.0`. Please implement the same change in all other templates.

**Comment by rcernich** [comment]
@vblagoje, @slaskawi, would it be better to put these into the image repository or a JDG specific repository?  Our longer term goal is to move image stream definitions and templates to product specific repositories.

**Comment by slaskawi** [comment]
@rcernich hmmm interesting. What do you think about putting it into https://github.com/jboss-container-images/jboss-datagrid-7-openshift-image into `templates` and `image-streams` directories? BTW, I believe we should also update some OpenShift Online configuration files and tell it where to find those templates, right? Could you please tell us what is the process for handling those?

**Comment by rcernich** [comment]
This is what the JDV folks did: https://github.com/jboss-container-images/jboss-datavirt-6-openshift-image/tree/datavirt64-dev/resources/openshift/templates

File: datagrid/datagrid72-image-stream.json [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+                            "description": "JBoss Data Grid 7.2 S2I images.",
+                            "iconClass": "icon-datagrid",
+                            "tags": "datagrid,jboss,hidden",
+                            "supports": "datagrid:7.2",
+                            "version": "1.0",
+                            "openshift.io/display-name": "Red Hat JBoss Data Grid 7.2"
+                        },
+                        "from": {
+                            "kind": "DockerImage",
+                            "name": "jboss-datagrid-7/datagrid72-openshift:1.0"
```
Comment (maschmid) [inline datagrid/datagrid72-image-stream.json]: should this contain the registry.access.redhat.com/  registry prefix?

**Comment by rcernich** [comment]
This PR should be closed, as the templates are collocated with the image source: jboss-container-images/jboss-datagrid-7-openshift-image#56

**Comment by osnipso** [comment]
Closing, as this is now outdated.

**Pull request closed without merging**